### PR TITLE
OpenCensus Cloud Spanner: capture GFE metric sample

### PR DESCRIPTION
**Pull request by mayurkale22**
Continuation of https://github.com/GoogleCloudPlatform/java-docs-samples/pull/5437 (This is the last PR in installment)

> It's a good idea to open an issue first for discussion.

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md)
- [x] `pom.xml` parent set to latest `shared-configuration`
- [x] Appropriate changes to README are included in PR
- [ ] API's need to be enabled to test (tell us)
- [ ] Environment Variables need to be set (ask us to set them)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [x] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [x] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: spanner/opencensus/src/main/java/com/example/spanner/opencensus/CaptureGfeMetric.java [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        System.out.printf(
+            "%d %d %s", resultSet.getLong(0), resultSet.getLong(1), resultSet.getString(2));
+      }
+    }
+  }
+
+  private static final HeaderClientInterceptor interceptor = new HeaderClientInterceptor();
+  private static final Metadata.Key<String> SERVER_TIMING_HEADER_KEY =
+      Metadata.Key.of("server-timing", Metadata.ASCII_STRING_MARSHALLER);
+  private static final Pattern SERVER_TIMING_HEADER_PATTERN = Pattern.compile(".*dur=(?<dur>\\d+)");
```
Comment (lesv) [inline spanner/opencensus/src/main/java/com/example/spanner/opencensus/CaptureGfeMetric.java:134]: OPTIONAL - consider mentioning where this came from.

File: spanner/opencensus/src/main/java/com/example/spanner/opencensus/CaptureGfeMetric.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+          super.start(new SimpleForwardingClientCallListener<RespT>(responseListener) {
+            @Override
+            public void onHeaders(Metadata metadata) {
+              processHeader(metadata, method.getFullMethodName());
+              super.onHeaders(metadata);
+            }
+          }, headers);
+        }
+      };
+    }
```
Comment (lesv) [inline spanner/opencensus/src/main/java/com/example/spanner/opencensus/CaptureGfeMetric.java:155]: OPTIONAL - consider explaining this or provide a link to the docs.  If it's not already on the page you display this on..

**Review by lesv — APPROVED** [review]
A couple of OPTIONAL suggestions, otherwise LGTM

**Comment by lesv** [comment]
Just ping me if you choose to ignore the suggestions, if you accept them, just ping me and I'll review and merge.

File: spanner/opencensus/src/main/java/com/example/spanner/opencensus/CaptureGfeMetric.java (hunk 1 above)
Comment (mayurkale22) [reply spanner/opencensus/src/main/java/com/example/spanner/opencensus/CaptureGfeMetric.java:136]: Done. Thanks

**Comment by mayurkale22** [comment]
> Just ping me if you choose to ignore the suggestions, if you accept them, just ping me and I'll review and merge.

Thanks for the reviews, I addressed them. PTAL.

**Review by lesv — APPROVED** [review]

**Pull request merged**